src/geometry/PrimitiveGroup.ts

```diff
@@ -102,20 +119,30 @@ export default abstract class PrimitiveGroup extends GeometryGroup {
   }
 
   setTreeIndex(value: number, index: number) {
-    if (this.treeIndexMap[value] == null) {
-      this.treeIndexMap[value] = [index];
-    } else {
-      this.treeIndexMap[value].push(index);
-    }
-
     this.treeIndex[index] = value;
   }
 
+  buildTreeIndexMap() {
+    this.treeIndexMap = {};
+    this.treeIndex.forEach((treeIndex, index) => {
+      if (index < this.data.count) {
```

Comment:
Can it ever happen that `index` is larger?

Reply:
Switched to a for loop